### PR TITLE
Add at' that works for all At instances

### PR DESCRIPTION
**Pull request by arybczak**
I had a moment of enlightenment 🧠

Whether using `at'` with `view` should force the value is debatable.

### CODE REVIEW

File: optics-core/src/Optics/At/Core.hs [hunk 1]
```diff
@@ -366,10 +368,45 @@ class Ixed m => At m where
   -- >>> at 1 ?~ "hello" $ Map.empty
   -- fromList [(1,"hello")]
   --
+  -- /Note:/ Usage of this function might introduce space leaks if you're not
+  -- careful to make sure that values put inside the 'Just' constructor are
+  -- evaluated. If this is the case, use 'at'' instead.
```
Comment (adamgundry) [inline optics-core/src/Optics/At/Core.hs]:
```suggestion
  -- evaluated. To force the values and avoid such leaks, use 'at'' instead.
```

File: optics-core/src/Optics/At/Core.hs [hunk 2]
```diff
@@ -366,10 +368,45 @@ class Ixed m => At m where
...
 
+-- | Version of 'at' strict in the value inside the `Just` constructor.
+--
+-- Example:
+--
+-- >>> (at () .~ Just (errorWithoutStackTrace "oops") $ Nothing) `seq` ()
+-- ()
+--
+-- >>> (at' () .~ Just (errorWithoutStackTrace "oops") $ Nothing) `seq` ()
+-- *** Exception: oops
```
Comment (adamgundry) [inline optics-core/src/Optics/At/Core.hs]: Is there any reason this couldn't just be `error`, and use `...` to satisfy doctest?
```
-- >>> (at' () .~ Just (error "oops") $ Nothing) `seq` ()
-- *** Exception: oops
-- ...
```

**Comment by arybczak** [comment]
> Interestingly this is stricter than _1' (should that be the case?)

`_1'` only forces the outer constructor, you force all fields - using `_1''` for accessing a tuple essentially makes it a strict tuple.

I guess that's the other problem of strict function variants heh - they introduce varying degree of strictness.

File: optics-core/src/Optics/At/Core.hs (hunk 2 above)
Comment (arybczak) [reply optics-core/src/Optics/At/Core.hs]: Oh, I didn't know that trick. Yes, doing that makes sense.

**Pull request merged**